wit/types.wit

```diff
@@ -16,29 +25,50 @@ interface messaging-types {
     /// The interface doesn't highlight this difference in the type itself as that's uniquely a consumer issue.
     type channel = string;
   
-    /// Configuration includes a required list of channels the guest is subscribing to, and an optional list of extensions key-value pairs 
-    /// (e.g., partitions/offsets to read from in Kafka/EventHubs, QoS etc.).
+    /// Configuration includes a required list of channels the guest is subscribing to, and an
+    /// optional list of extensions key-value pairs (e.g., partitions/offsets to read from in 
+    /// Kafka/EventHubs, QoS etc.).
     record guest-configuration {
         channels: list<channel>,
         extensions: option<list<tuple<string, string>>>
     }
   
-    /// Format specification for messages 
-    ///  - more info: https://github.com/clemensv/spec/blob/registry-extensions/registry/spec.md#message-formats
-    ///  - message metadata can further decorate w/ things like format version, and so on.
-    enum format-spec {
-        cloudevents,
-        http,
-        amqp,
-        mqtt,
-        kafka,
-        raw
-    }
-  
-    /// A message with a binary payload, a format specification, and decorative metadata.
-    record message {
-        data: list<u8>,
-        format: format-spec,
-        metadata: option<list<tuple<string, string>>>
+    /// A message with a binary payload and additional information
+    resource message {
```

Comment:
I think it would be great to have some additional methods here for adding additional fields to the metadata / overriding the topic in the case of a message proxy